src/Microsoft.AspNet.Http/DefaultHttpResponse.cs

```diff
@@ -54,21 +56,20 @@ public override Stream Body
 
         public override long? ContentLength
         {
-            get
-            {
-                return ParsingHelpers.GetContentLength(Headers);
-            }
-            set
-            {
-                ParsingHelpers.SetContentLength(Headers, value);
-            }
+            get { return HttpResponseFeature.ContentLength; }
```

Comment:
No; not with the code here; however it does rely on the ```engine/server``` implementer of ```IHttpResponseFeature``` maintaining this behaviour. Middleware wrapping that implementation will still be safe.